dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/persistence/Serializer.kt

```diff
@@ -15,4 +18,30 @@ internal interface Serializer<T : Any> {
      * @return the String representing the data or null if any exception occurs
      */
     fun serialize(model: T): String?
+
+    companion object {
+        const val ERROR_SERIALIZING = "Error serializing %s model"
+    }
+}
+
+/**
+ * A utility class to serialize a model to a ByteArray safely.
+ * If an exception is thrown while serializing the data, null is returned, and a
+ * message will be sent to the internalLogger.
+ */
+@Suppress("TooGenericExceptionCaught")
+internal fun <T : Any> Serializer<T>.serializeToByteArray(
+    model: T,
+    internalLogger: Logger
+): ByteArray ? {
+    return try {
+        val serialized = serialize(model)
+        serialized?.toByteArray(Charsets.UTF_8)
+    } catch (e: Throwable) {
+        internalLogger.e(
+            Serializer.ERROR_SERIALIZING.format(Locale.US, model.javaClass.simpleName),
+            e
+        )
+        null
+    }
```

Comment:
Just a sanity check, as I can't see this from wider context - is this logic excluded from serializing `internalLogger` events itself? It's necessary to avoid the situation where `internalLogger` produces a bad `model` and then it sends serialization error, which leads to another bad `model` ... ♾️.

Reply:
Yes it is, the internal logger is provided in the constructors, and it'll be a logger that only prints to dev console for internal monitoring events